src/tensor_ops/prelu/mod.rs

```diff
@@ -0,0 +1,116 @@
+use core::fmt::Debug;
+
+use num_traits::Zero;
+
+use crate::{shapes::*, tensor::*};
+
+use super::{
+    cmp::{LtKernelOp, ScalarCmpKernel},
+    BroadcastTo, ChooseFrom, Device,
+};
+
+#[repr(C)]
+#[derive(Debug, Default, Clone, Copy)]
+pub struct PReLUKernelOp;
+
+#[repr(C)]
+#[derive(Debug, Default, Clone, Copy)]
+pub struct LeakyReLUKernelOp<E> {
+    slope: E,
+}
+
+/// [Parametric Rectified Linear Unit (PReLU)](https://pytorch.org/docs/stable/generated/torch.nn.PReLU.html). `max(0, t) + a*min(0, t)`
+///
+/// Examples:
+/// ```rust
+/// # use dfdx::prelude::*;
+/// # let dev: Cpu = Default::default();
+/// let t = dev.tensor([-1.0, 0.0, 1.0, 2.0]);
+/// let a = dev.tensor([0.05,0.05,0.05,0.05]);
+/// let r = prelu(t, a);
+/// assert_eq!(r.array(), [-0.05, 0.0, 1.0, 2.0]);
+/// ```
+
+pub fn prelu<S: Shape, E: Dtype, D: Device<E>, T: Tape<E, D> + Merge<R>, R: Default>(
+    lhs: Tensor<S, E, D, T>,
+    rhs: Tensor<S, E, D, R>,
+) -> Tensor<S, E, D, T>
+where
+    Tensor<S, E, D, T>: TryPReLU<Tensor<S, E, D, R>, Output = Tensor<S, E, D, T>>,
+{
+    lhs.prelu(rhs)
+}
+
+pub fn leakyrelu<S: Shape, E: Dtype, D: Device<E>, T: Tape<E, D>>(
+    lhs: Tensor<S, E, D, T>,
+    rhs: E,
+) -> Tensor<S, E, D, T>
+where
+    Tensor<S, E, D, T>: TryPReLU<E, Output = Tensor<S, E, D, T>>,
+{
+    lhs.prelu(rhs)
+}
+
+pub trait TryPReLU<T = Self>: HasErr {
+    type Output;
+
+    fn try_prelu(self, rhs: T) -> Result<Self::Output, Self::Err>;
+
+    fn prelu(self, rhs: T) -> Self::Output {
+        self.try_prelu(rhs).unwrap()
+    }
```

Comment:
This did not work, I'm guessing that it is because the bound of ScalarCmpKernel<LtKernelOp, E> is not part of Device<E>. I did remove the assoc trait Output.